promql/ast.go

```diff
@@ -231,6 +231,9 @@ type VectorMatching struct {
 	// On contains the labels which define equality of a pair
 	// of elements from the vectors.
 	On model.LabelNames
+	// Ignoring excludes the given label names from matching,
+	// rather than only using them.
+	Ignoring bool
```

Comment:
If `Ignoring` is true, the name `On` is very misleading. Also, we have developed a certain aversion against binary flags in function signatures (which happens when using the `Ignoring` bool.

Idea: How about having an `Ignoring model.LabelNames` in addition to `On model.LabelNames` with the invariant that at least one of them must have a length of 0.

But I leave final judgement to @fabxc as the master of the engine.